src/runtime/converter.cs

```diff
@@ -135,8 +135,16 @@ internal static IntPtr ToPython(object value, Type type)
                 return result;
             }
 
-			if (value is IList && !(value is INotifyPropertyChanged) && value.GetType().IsGenericType)
-			{
+            if (Type.GetTypeCode(type) == TypeCode.Object && value.GetType() != typeof(object)) {
```

Comment:
glad this comes before other builtin conversions but ultimately the built-in conversions should probably use this codec system internally.